molgenis-omx-protocolviewer/src/main/java/org/molgenis/omx/protocolviewer/ProtocolViewerController.java

```diff
@@ -247,12 +247,22 @@ public String getOrderDataForm()
 	// Spring's StandardServletMultipartResolver can't bind a RequestBody or
 	// ModelAttribute
 	@RequestMapping(value = "/order", method = RequestMethod.POST, headers = "Content-Type=multipart/form-data")
-	@ResponseStatus(HttpStatus.NO_CONTENT)
-	public void orderData(@RequestParam Integer catalogId, @RequestParam String name, @RequestParam Part file)
+	public ModelAndView orderData(@RequestParam Integer catalogId, @RequestParam String name, @RequestParam Part file)
 			throws IOException, MessagingException, UnknownCatalogException, UnknownStudyDefinitionException
 	{
 		if (!getEnableOrderAction()) throw new MolgenisDataAccessException("Action not allowed");
-		protocolViewerService.submitStudyDefinitionDraftForCurrentUser(name, file, catalogId.toString());
+		try {
+			protocolViewerService.submitStudyDefinitionDraftForCurrentUser(name, file, catalogId.toString());
+			ModelAndView model = new ModelAndView("order-response");
+			model.addObject("ok", true);
+			model.addObject("message", "Your submission has been received.");
+			return model;
+		} catch (Exception ex){
```

Comment:
The IFrame solution isn't able to access the HTTP status codes, and will always call the complete() callback. So the idea was to put the response in a success message instead.